app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt

```diff
@@ -64,12 +67,48 @@ class SplashActivityPresenter @Inject constructor(
     subscribeToOnboardingFlow()
   }
 
+  /** Handles cases where the user clicks the close app option on a deprecation notice dialog. */
   fun handleOnCloseAppButtonClicked() {
     // If the app close button is clicked for the deprecation notice, finish the activity to close
     // the app.
     activity.finish()
   }
 
+  /** Handle cases where the user clicks the update option on a deprecation notice dialog. */
+  fun handleOnUpdateButtonClicked() {
+    // If the Update button is clicked for the deprecation notice, launch the Play Store and open
+    // the Oppia app's page.
+
+    // Launch profile selection activity in case the user presses the back button
+    activity.startActivity(ProfileChooserActivity.createProfileChooserActivity(activity))
+    activity.finish()
```

Comment:
It is unclear why this function handles two unrelated actions. Could you split them, or clarify first why this is the case?

Reply:
What if the user clicks back, but the update is not yet installed(force update scenario)? I would suggest finishing the Splash activity in that scenario so that the app is closed/back to launcher for that scenario, since the user shouldn't be able to access the app.

Reply:
I have updated the action to close the app instead of navigating to a new activity. You can check the attached videos above.